cmd/talosctl/cmd/talos/version.go

```diff
@@ -82,8 +97,12 @@ var versionCmd = &cobra.Command{
 }
 
 func init() {
-	versionCmd.Flags().BoolVar(&shortVersion, "short", false, "Print the short version")
-	versionCmd.Flags().BoolVar(&clientOnly, "client", false, "Print client version only")
+	versionCmd.Flags().BoolVar(&versionCmdFlags.shortVersion, "short", false, "Print the short version")
+	versionCmd.Flags().BoolVar(&versionCmdFlags.clientOnly, "client", false, "Print client version only")
+
+	// TODO remove when https://github.com/talos-systems/talos/issues/907 is implemented
+	versionCmd.Flags().BoolVar(&versionCmdFlags.json, "json", false, "")
+	cli.Should(versionCmd.Flags().MarkHidden("json"))
```

Comment:
The alternative is to call an API in the CLI integration test that sounds strange